src/ReactiveDomain.Testing/Messaging/ConcurrentMessageQueue.cs

```diff
@@ -57,7 +58,10 @@ public ConcurrentMessageQueue(string name, IEnumerable<T> items)
         public void AssertEmpty()
         {
             if (!IsEmpty)
-                throw new Exception($" {_name} Queue not Empty.");
+            {
+                var msg = this.ElementAt(0);
```

Comment:
Let's use `TryPeek` for this since it's a core method in `ConcurrentQueue` rather than relying on a LINQ method.